extensions/Webhooks/lib/API/V1/Webhooks.pm

```diff
@@ -0,0 +1,63 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::Webhooks::API::V1::Webhooks;
+
+use 5.10.1;
+use Mojo::Base qw( Mojolicious::Controller );
+
+use Bugzilla::Constants;
+use Bugzilla::Extension::Webhooks::Webhook;
+
+use Mojo::JSON qw(true false);
+
+sub setup_routes {
+  my ($class, $r) = @_;
+  my $routes = $r->under(
+    '/webhooks' => sub { Bugzilla->usage_mode(USAGE_MODE_MOJO_REST); });
+  $routes->get('/list')->to('Webhooks::API::V1::Webhooks#list');
+}
+
+sub list {
+  my $self = shift;
+  Bugzilla->usage_mode(USAGE_MODE_MOJO_REST);
+  my $user = $self->bugzilla->login;
+  $user->id || return $self->user_error('login_required');
+
+  my $webhooks
+    = Bugzilla::Extension::Webhooks::Webhook->match({user_id => $user->id});
+
+  my @results;
+  foreach my $webhook (@{$webhooks}) {
+    push @results, $self->_webhook_to_hash($webhook);
+  }
+
+  return $self->render(json => {webhooks => \@results});
+}
+
+sub _webhook_to_hash {
+  my ($self, $webhook) = @_;
+
+  my $connector
+    = Bugzilla->push_ext->connectors->by_name('Webhook_' . $webhook->id);
```

Comment:
$webhook->name is not the same as the push connector name. $webhook->name is the descriptive name the user gave to their webhook such as "Send notifications to some server when some product changes". Webhook_1 or any other value for ID is the dynamic name that the push code generates when storing a push event in the DB.

Reply:
Okay, makes sense.

Out of scope for this change, but we ought to have a function that returns a valid webhook name, or maybe a `->by_id()` constructor so we don't need to hardcode `Webhook_ + id` everywhere. The function could just call `by_name('Webook_' . id)` but it would avoid a footgun in the long run.

Reply:
Good point.